notification_controller/controller/controller.go

```diff
@@ -71,16 +78,30 @@ func NewController(
 		appInformer:       appInformer,
 		appProjInformer:   appProjInformer,
 		apiFactory:        apiFactory}
-	res.ctrl = controller.NewController(appClient, appInformer, apiFactory,
-		controller.WithSkipProcessing(func(obj v1.Object) (bool, string) {
-			app, ok := (obj).(*unstructured.Unstructured)
-			if !ok {
-				return false, ""
-			}
-			return !isAppSyncStatusRefreshed(app, log.WithField("app", obj.GetName())), "sync status out of date"
-		}),
-		controller.WithMetricsRegistry(registry),
-		controller.WithAlterDestinations(res.alterDestinations))
+
+	if !selfServiceNotificationEnabled {
+		res.ctrl = controller.NewController(appClient, appInformer, apiFactory,
+			controller.WithSkipProcessing(func(obj v1.Object) (bool, string) {
+				app, ok := (obj).(*unstructured.Unstructured)
+				if !ok {
+					return false, ""
+				}
+				return !isAppSyncStatusRefreshed(app, log.WithField("app", obj.GetName())), "sync status out of date"
+			}),
+			controller.WithMetricsRegistry(registry),
+			controller.WithAlterDestinations(res.alterDestinations))
+	} else {
+		res.ctrl = controller.NewControllerWithNamespaceSupport(appClient, appInformer, apiFactory,
+			controller.WithSkipProcessing(func(obj v1.Object) (bool, string) {
+				app, ok := (obj).(*unstructured.Unstructured)
+				if !ok {
+					return false, ""
+				}
+				return !isAppSyncStatusRefreshed(app, log.WithField("app", obj.GetName())), "sync status out of date"
+			}),
+			controller.WithMetricsRegistry(registry),
+			controller.WithAlterDestinations(res.alterDestinations))
```

Comment:
Can we pull these 3 out as vars, since they seem to be the same between both conditions?